pkg/datalayer/datalayer.go

```diff
@@ -1,28 +1,172 @@
 package datalayer
 
 import (
+	"encoding/json"
+	"fmt"
 	"os"
+	"os/exec"
 	"path/filepath"
+	"strings"
+
+	"github.com/nu7hatch/gouuid"
 )
 
 // ClusterHQ data layer, naive vfs (directory-based) implementation
 
 type DataLayer struct {
-	BasePath string
+	basePath string
+}
+
+type CommitId string
+type CommitMessage string
+
+type Commit struct {
+	Id      CommitId      `json:"id"`
+	Message CommitMessage `json:"message"`
+}
+
+func NewDataLayer(basePath string) *DataLayer {
```

Comment:
I wasn't referring to that - I meant the tactic of implementing an OO-style constructor.